dfinity/nns-dapp
# Store SNS accounts in icrcAccountsStore

# Motivation

The ICRC wallet loads accounts into `icrcAccountsStore` keyed by `ledgerCanisterId`.
Since we started using the `IcrcWalletPage` component for the `SnsWallet` those accounts are now also stored there.
But we still also have the `snsAccountsStore` in which accounts are loaded by other components.
This means SNS accounts are loaded twice an can be inconsistent.
We should always load SNS accounts into the `icrcAccountsStore`.

# Changes

1. Change `loadSnsAccounts` to call `loadAccounts` which loads accounts into `icrcAccountsStore` keyed by `ledgerCanisterId`.
2. Remove `snsAccountsStore` and make a new derived `snsAccountsStore` with SNS accounts keyed by `rootCanisterId` for convenience and to not have to change all existing code. The types are copied over from the original store.
3. Remove `snsAccountsStore` from the debug store as it's now a derived store.
4. Change `universesAccountsStore` to make sure SNS accounts are keyed by universe/rootCanisterId and are not duplicated.

# Tests

1. Add unit test for derived `snsAccountsStore`.
2. Update existing tests to set up SNS projects and `icrcAccountsStore` or `queryIcrcBalance`, such that the `snsAccountsStore` has the expected state.
3. Remove `mockSnsAccountsStoreSubscribe` which is now unused.
4. Tested manually at https://qsgjb-riaaa-aaaaa-aaaga-cai.dskloet-ingress.devenv.dfinity.network/

# Todos
